### ISSUE TITLE
1.22.0 Custom docker-compose config can't use port 9999 due to conflict with traefik management port

### ISSUE THREAD

**Issue by tomderoo** [ddev/ddev#5196]
### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
Running bash [-c /tmp/test_ddev.sh] 
======= Existing project config =========
These config files were loaded for project myproject: [/home/myuser/ddev/myproject/.ddev/config.yaml /home/myuser/ddev/myproject/.ddev/config.wkhtmltox.yaml]
name: myproject
type: php
docroot: public
php_version: 8.2
webserver_type: apache-fpm
webimage: ddev/ddev-webserver:v1.22.0
router_http_port: 80
router_https_port: 443
database: {mariadb 10.6}
hooks: map[post-start:[map[exec:mysql -uroot -proot -hdb -e "CREATE DATABASE IF NOT EXISTS db_test; GRANT ALL ON db_test.* TO 'db'@'%';"]]]
mailhog_port: 8025
mailhog_https_port: 8026
webimage_extra_packages: [php8.2-mailparse xfonts-75dpi xfonts-base]
project_tld: ddev.site
use_dns_when_possible: true
composer_version: 2
nodejs_version: 16
default_container_timeout: 120
======= Creating dummy project named  tryddevproject-8756 in ../tryddevproject-8756 =========
OS Information: Linux myuser-ThinkPad-T14-Gen-3 5.19.0-46-generic #47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 21 15:35:31 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
User information: uid=1000(myuser) gid=1000(myuser) groups=1000(myuser),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),122(lpadmin),135(lxd),136(sambashare),139(libvirt),999(docker)
DDEV version:  ITEM             VALUE                                   
 DDEV version     v1.22.0                                 
 architecture     amd64                                   
 db               ddev/ddev-dbserver-mariadb-10.4:v1.22.0 
 ddev-ssh-agent   ddev/ddev-ssh-agent:v1.22.0             
 docker           24.0.5                                  
 docker-compose   v2.20.0                                 
 docker-platform  Ubuntu 22.04.2 LTS                      
 mutagen          0.17.2                                  
 os               linux                                   
 router           traefik:v2.10                           
 web              ddev/ddev-webserver:v1.22.0             
PROXY settings: HTTP_PROXY='' HTTPS_PROXY='' http_proxy='' NO_PROXY=''
======= DDEV global info =========
Global configuration:
instrumentation-opt-in=true
omit-containers=[]
performance-mode=none
router-bind-all-interfaces=false
internet-detection-timeout-ms=3000
disable-http2=false
use-letsencrypt=false
letsencrypt-email=
table-style=default
simple-formatting=false
auto-restart-containers=false
use-hardened-images=false
fail-on-hook-fail=false
required-docker-compose-version=v2.20.0
use-docker-compose-from-path=false
project-tld=
xdebug-ide-location=
no-bind-mounts=false
router=traefik
wsl2-no-windows-hosts-mgt=false
router-http-port=80
router-https-port=443

======= DOCKER info =========
docker location: -rwxr-xr-x 1 root root 35927416 jul 21 22:35 /usr/bin/docker
docker version: 
Client: Docker Engine - Community
 Version:           24.0.5
 API version:       1.43
 Go version:        go1.20.6
 Git commit:        ced0996
 Built:             Fri Jul 21 20:35:18 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.5
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.6
  Git commit:       a61e2b4
  Built:            Fri Jul 21 20:35:18 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
DOCKER_DEFAULT_PLATFORM=notset
======= Mutagen Info =========
======= Docker Info =========
Docker platform: Ubuntu 22.04.2 LTS 
Using docker context: default (unix:///var/run/docker.sock) 
docker-compose: v2.20.0 
Using DOCKER_HOST=unix:///var/run/docker.sock 
Docker version: 24.0.5 
Able to run simple container that mounts a volume. 
Able to use internet inside container. 
Docker disk space:
Filesystem                Size      Used Available Use% Mounted on
overlay                 913.8G    263.6G    603.7G  30% /

The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again.
Existing docker containers: 
CONTAINER ID   IMAGE          COMMAND                  CREATED        STATUS                    PORTS     NAMES
8f16f7b2495c   7a546d03c39d   "/bin/sh"                2 weeks ago    Created                             phpstorm_helpers_PS-231.9161.47
6dbfbafa35a4   ea2c691c59b1   "/bin/sh"                2 months ago   Created                             phpstorm_helpers_PS-231.9011.38
4b15883d5d2b   ba24e115912e   "/bin/sh"                3 months ago   Created                             phpstorm_helpers_PS-231.8109.199
b6298d47d7d2   2986bb9a7b89   "/bin/sh"                4 months ago   Created                             phpstorm_helpers_PS-223.8836.42
2d4a9a160389   60ad69959caa   "/bin/sh"                4 months ago   Created                             phpstorm_helpers_PS-223.8617.59
b569ae9e043f   lawcloud       "/entrypoint supervi…"   4 months ago   Exited (0) 4 months ago             epic_carson
ca83b0b18670   hello-world    "/hello"                 4 months ago   Exited (0) 4 months ago             beautiful_swartz
39dadc1f859e   hello-world    "/hello"                 4 months ago   Exited (0) 4 months ago             vibrant_spence
Network ddev_default created 
Starting tryddevproject-8756... 
 Container ddev-ssh-agent  Created 
 Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
 Network ddev-tryddevproject-8756_default  Created 
 Container ddev-tryddevproject-8756-web  Created 
 Container ddev-tryddevproject-8756-db  Created 
 Container ddev-tryddevproject-8756-web  Started 
 Container ddev-tryddevproject-8756-db  Started 
 Container ddev-router  Created 
 Container ddev-router  Started 
Successfully started tryddevproject-8756 
Project can be reached at https://tryddevproject-8756.ddev.site https://127.0.0.1:32866 
======== Curl of site from inside container:
HTTP/1.1 200 OK
Server: nginx
Date: Wed, 26 Jul 2023 08:53:20 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== curl -I of http://tryddevproject-8756.ddev.site from outside:
HTTP/1.1 200 OK
Content-Type: text/html; charset=UTF-8
Date: Wed, 26 Jul 2023 08:53:21 GMT
Server: nginx
Vary: Accept-Encoding

======== full curl of http://tryddevproject-8756.ddev.site from outside:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-8756'
======== Project ownership on host:
drwxr-xr-x 4 tomderoo tomderoo 4096 jul 26 10:53 ../tryddevproject-8756
======== Project ownership in container:
drwxr-xr-x 4 tomderoo tomderoo 4096 Jul 26 08:53 /var/www/html
======== In-container filesystem:
Filesystem                Type 1K-blocks      Used Available Use% Mounted on
/dev/mapper/vgubuntu-root ext4 958137496 276569416 632823632  31% /var/www/html
======== curl again of tryddevproject-8756 from host:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-8756'
Thanks for running the diagnostic. It was successful.
```
</details>         

### Expected Behavior

After updating to 1.22.0, I expect `ddev start` to actually start my DDEV project, but instead I get a conflict on port 9999: `services.ddev-router.ports array items[5,6] must be unique`

#### **Update:** There are two easy fixes:
1. Go back to the traditional nginx-proxy router, which does not use port 9999: `ddev config global --router=nginx-proxy`
2. Change your `docker-compose.*.yaml` to use another port than port 9999.  This requires changing HTTP_EXPOSE and HTTPS_EXPOSE to use an alternate exposed port.

### Actual Behavior

After upgrading to 1.22.0, I receive the following error on starting my project:

```
validating /home/myuser/.ddev/.router-compose.yaml: services.ddev-router.ports array items[5,6] must be unique 
Failed to start myproject: ComposeCmd failed to run 'COMPOSE_PROJECT_NAME=ddev-myproject docker-compose -f /home/myuser/.ddev/.router-compose.yaml config', action='[config]', err='exit status 15', stdout='', stderr='validating /home/myuser/.ddev/.router-compose.yaml: services.ddev-router.ports array items[5,6] must be unique' 
```

The cause is pretty obvious when checing the contents of `/home/myuser/.ddev/.router-compose.yaml`:

```yaml
services:
  ddev-router:
    image: traefik:v2.10-built
    build:
      context: router-build
      args:
        BASE_IMAGE: 'traefik:v2.10'
        username: 'myuser'
        uid: '1000'
        gid: '1000'

    
    command:
      - --configFile=/mnt/ddev-global-cache/traefik/static_config.yaml
    user: 1000:1000
    

    networks:
        - ddev_default
    container_name: ddev-router
    ports:
    - "127.0.0.1:443:443"
    - "127.0.0.1:80:80"
    - "127.0.0.1:8025:8025"
    - "127.0.0.1:8026:8026"
    # MY COMMENT: here starts the ugly part; these ports are added because of ddev/mercure config
    - "127.0.0.1:9998:9998"
    - "127.0.0.1:9999:9999"
    
    # Traefik router; configured in static config as entrypoint
    # TODO: Make this configurable? Put it somewhere else?
    # MY COMMENT: ... and you will notice that Traefik auto-adds the exact same port as Mercure did above
    - "127.0.0.1:9999:9999"
    
    volumes:
      
      - ddev-global-cache:/mnt/ddev-global-cache:rw
       
    environment:
      - DISABLE_HTTP2=false
        
    restart: "no"
    healthcheck:
      
      test: "/healthcheck.sh"
      
      interval: 1s
      retries: 120
      start_period: 120s
      timeout: 120s

networks:
  ddev_default:
    name: ddev_default
    external: true
volumes:
  ddev-global-cache:
    name: ddev-global-cache
    external: true
```

The port config `"127.0.0.1:9999:9999"` is added twice. The first time comes from the `mercure` docker compose copied from the Mercure Symfony bundle, the second one comes from Traefik.

Here is my project's `docker-compose.mercure.yaml`:

```yaml
version: '3'

services:
  mercure:
    image: dunglas/mercure
    restart: unless-stopped
    container_name: "ddev-${DDEV_SITENAME}-mercure-hub"
    labels:
      com.ddev.site-name: ${DDEV_SITENAME}
      com.ddev.approot: ${DDEV_APPROOT}
    expose:
      - "3000"
    environment:
      VIRTUAL_HOST: $DDEV_HOSTNAME
      SERVER_NAME: ":3000"
      HTTP_EXPOSE: "9998:3000"
      HTTPS_EXPOSE: "9999:3000"
      MERCURE_PUBLISHER_JWT_KEY: '1289c268ad0f83eb51036592213bf45794e03c969907dac0d622b3ea22a6d3ab'
      MERCURE_SUBSCRIBER_JWT_KEY: '1289c268ad0f83eb51036592213bf45794e03c969907dac0d622b3ea22a6d3ab'
      MERCURE_EXTRA_DIRECTIVES: |
        cors_origins https://myproject.ddev.site
    # Comment the following line to disable the development mode
    command: /usr/bin/caddy run --config /etc/caddy/Caddyfile.dev
    volumes:
      - mercure_data:/data
      - mercure_config:/config

volumes:
  mercure_data:
  mercure_config:
```
I would be more than happy to simply change the `HTTP_EXPOSE` and `HTTPS_EXPOSE` values, but then Mercure stops working in my Symfony app due to a 502 bad gateway / cors origin on the selected port (I tried the HTTPS one with 9990, I assume it fails with anything that is not 9999), even after changing the `.env` to have the mercure related variables match the new ports. I do not know why it fails on anything that is not 9999, I am not versed in Mercure config.

I can also **avoid the problem by stepping away from Traefik and reconfiguring ddev globally to use the old nginx-proxy router** through `ddev config global --router=nginx-proxy`, while keeping the "working" mercure config. My `/home/myuser/.ddev/.router-compose.yaml` then outputs:

```yaml
services:
  ddev-router:
    image: ddev/ddev-router:v1.22.0-built
    build:
      context: router-build
      args:
        BASE_IMAGE: 'ddev/ddev-router:v1.22.0'
        username: 'myuser'
        uid: '1000'
        gid: '1000'

    

    networks:
        - ddev_default
    container_name: ddev-router
    ports:
    - "127.0.0.1:443:443"
    - "127.0.0.1:80:80"
    - "127.0.0.1:8025:8025"
    - "127.0.0.1:8026:8026"
    - "127.0.0.1:9998:9998"
    - "127.0.0.1:9999:9999"
    
    volumes:
      
      - /var/run/docker.sock:/tmp/docker.sock:ro
      
      - ddev-global-cache:/mnt/ddev-global-cache:rw
       
    environment:
      - DISABLE_HTTP2=false
        
    restart: "no"
    healthcheck:
      
      interval: 1s
      retries: 120
      start_period: 120s
      timeout: 120s

networks:
  ddev_default:
    name: ddev_default
    external: true
volumes:
  ddev-global-cache:
    name: ddev-global-cache
    external: true
```

You will notice: no duplicate ports.

### Steps To Reproduce

1. Have ddev 1.22.0
3. Have a ddev project with docker mercure config from the Mercure Symfony bundle on the standard config (with specifically the https port at 9999)
4. Make sure the ddev router is traefik (the default since 1.22.0)
5. Try to start your project, see it fail

**Comment by rfay** [comment]
You've done a great job studying this. Could you please create a trivial example that demonstrates the problem? It seems like it must have nothing to do with mercure, but I'd never heard of mercure before and am not necessarily following what you're up to.

If you can just create a trivial PHP project (nothing but an index.php) with a .ddev config and docker-compose.breakit.yaml that shows this it would be much appreciated. 

Possiblly related: https://discord.com/channels/664580571770388500/1133658524967260252/1133658524967260252

**Comment by tomderoo** [comment]
> It seems like it must have nothing to do with mercure, but I'd never heard of mercure before and am not necessarily following what you're up to.

I believe you are right, in my example it is just related to Mercure simply because Mercure's docker compose is injecting the same port config as Traefik does, i.e. `9999:9999`. Any other docker compose that proposes the same port will run into the same problem.

So I think the crux of the problem is this part of `/home/myuser/.ddev/.router-compose.yaml`:

```yaml
    # Traefik router; configured in static config as entrypoint
    # TODO: Make this configurable? Put it somewhere else?
    - "127.0.0.1:9999:9999"
```

Whoever added that `TODO` was on to something; if I could just manage to configure that to some other port as entrypoint, then my problems would be solved. I think. But I have no idea how Traefik works, or if that port can be changed at all without breaking stuff...

**Comment by stasadev** [comment]
The easiest way to reproduce it is to add this to `.ddev/config.yaml` and start the project:

```yaml
web_extra_exposed_ports:
    - name: frontend
      container_port: 9999
      http_port: 9998
      https_port: 9999
```

**Comment by rfay** [comment]
Oh yeah, you won't be able to use port 9999, because it conflicts with the traefik port. You'll have to use a different port.

**Comment by pt1602** [comment]
Here the requested [example](https://github.com/pt1602/ddevIssueDemo). Hope it helps!

**Comment by stasadev** [comment]
The docs need to be updated: there are examples for port 9999 (just choose a different port as an example):
https://ddev.readthedocs.io/en/latest/users/extend/custom-compose-files/#docker-composeyaml-examples
https://ddev.readthedocs.io/en/latest/users/extend/customization-extendibility/#exposing-extra-ports-via-ddev-router

The hardest part is choosing a port because people will be copying from the docs.

I believe the reason for using 9999 in this issue is simply a copy-paste technique.

**Comment by rfay** [comment]
It would have been better to use a different port for traefik admin port :)

**Comment by rfay** [comment]
In your case @tomderoo just change from
```
      HTTP_EXPOSE: "9998:3000"
      HTTPS_EXPOSE: "9999:3000"
```

to
```
      HTTP_EXPOSE: "10001:3000"
      HTTPS_EXPOSE: "10002:3000"
```

**Comment by rfay** [comment]
TODO:
- [ ] Fix docs to not show 9999 in example
- [ ] Consider detecting and warning about use of 9999

**Comment by tomderoo** [comment]
> The docs need to be updated: there are examples for port 9999 (just choose a different port as an example): https://ddev.readthedocs.io/en/latest/users/extend/custom-compose-files/#docker-composeyaml-examples https://ddev.readthedocs.io/en/latest/users/extend/customization-extendibility/#exposing-extra-ports-via-ddev-router
> 
> The hardest part is choosing a port because people will be copying from the docs.
> 
> I believe the reason for using 9999 in this issue is simply a copy-paste technique.

Alright, I am not sure I understand all that correctly, but I read it as though I could simply configure any other port in this particular custom docker compose, i.e. `docker-compose.mercure.yanl`, to for instance:

```yaml
expose:
  - "3000"
environment:
  VIRTUAL_HOST: $DDEV_HOSTNAME
  SERVER_NAME: ":3000"
  HTTP_EXPOSE: "9898:3000"
  HTTPS_EXPOSE: "9899:3000"
```

But alas, **while this actually works with the deprecated `nginx-proxy` router, it does not with `traefik`: my ddev project starts (because there are no port duplications anymore), but I get cors errors in my browser network inspection and a 502 bad gateway when visiting that url directly**. It seems I need some other extra config in the Traefik ecosystem to allow those ports to work...

@rfay changing to `10001` and `10002` does not work either => same cors and 502 issue...

**Comment by rfay** [comment]
If you're having CORS issues they are application problems; please look farther there. I imagine you have to tell mercure that you are allowed to do this.

**Comment by tomderoo** [comment]
> If you're having CORS issues they are application problems; please look farther there. I imagine you have to tell mercure that you are allowed to do this.

CORS in the inspection netwark tab quite often means that there is "something wrong with the response" - it is the "MySQL server has gone away" of browser inspection world, these days :(

The important factor is the 502 I get on direct access when using `traefik`, while I can simply reach the url correctly when using the deprecated `nginx-proxy` router. (This in a context where I actually changed the ports in the docker compose for mercure.) So this being the only difference, I am guessing Traefik is doing "something" to prevent those other ports from working to connect. And I will repeat: I know absolutely nothing about Traefik, or the old ddev router for that matter, I have no idea what could cause this.

**Comment by rfay** [comment]
Do the alternate ports work without CORS issues using the traditional nginx-proxy router?

If they work there and don't work with traefik, then we've moved on to a completely different issue.

**Comment by tomderoo** [comment]
> Do the alternate ports work without CORS issues using the traditional nginx-proxy router?
> 
> If they work there and don't work with traefik, then we've moved on to a completely different issue.

Indeed they do.

With this global ddev config:

```
Global configuration: 
instrumentation-opt-in=true 
omit-containers=[] 
web-environment=[] 
performance-mode=none 
router-bind-all-interfaces=false 
internet-detection-timeout-ms=3000 
disable-http2=false 
use-letsencrypt=false 
letsencrypt-email= 
table-style=default 
simple-formatting=false 
auto-restart-containers=false 
use-hardened-images=false 
fail-on-hook-fail=false 
required-docker-compose-version=v2.20.0 
use-docker-compose-from-path=false 
project-tld= 
xdebug-ide-location= 
no-bind-mounts=false 
router=nginx-proxy 
wsl2-no-windows-hosts-mgt=false 
router-http-port=80 
router-https-port=443 
```

And this `docker-compose.mercure.yaml`:

```yaml
version: '3'

services:
  mercure:
    image: dunglas/mercure
    restart: unless-stopped
    container_name: "ddev-${DDEV_SITENAME}-mercure-hub"
    labels:
      com.ddev.site-name: ${DDEV_SITENAME}
      com.ddev.approot: ${DDEV_APPROOT}
    expose:
      - "3000"
    environment:
      VIRTUAL_HOST: $DDEV_HOSTNAME
      SERVER_NAME: ":3000"
      HTTP_EXPOSE: "10001:3000"
      HTTPS_EXPOSE: "10002:3000"
      MERCURE_PUBLISHER_JWT_KEY: '1289c268ad0f83eb51036592213bf45794e03c969907dac0d622b3ea22a6d3ab'
      MERCURE_SUBSCRIBER_JWT_KEY: '1289c268ad0f83eb51036592213bf45794e03c969907dac0d622b3ea22a6d3ab'
      MERCURE_EXTRA_DIRECTIVES: |
        cors_origins https://myproject.ddev.site
    # Comment the following line to disable the development mode
    command: /usr/bin/caddy run --config /etc/caddy/Caddyfile.dev
    volumes:
      - mercure_data:/data
      - mercure_config:/config

volumes:
  mercure_data:
  mercure_config:
```

Everything works fine on the `nginx-proxy` router. It is only when I change router to `traefik` that I get 502 on the mercure url (which in my case is `https://myproject.ddev.site:10002/.well-known/mercure`, which is pretty standard).

**Comment by rfay** [comment]
Please open a new issue, thanks. Please explain exactly what happens and provide a reproduction scenario. 

This issue was super valuable but now we're in a completely different world. And we'll attack the things discovered in the next point release.

**Comment by tomderoo** [comment]
> Please open a new issue, thanks. Please explain exactly what happens and provide a reproduction scenario.
> 
> This issue was super valuable but now we're in a completely different world. And we'll attack the things discovered in the next point release.

Um, I would like to, but I am not sure how to proceed on that. As I mentioned in my original post:

> I would be more than happy to simply change the `HTTP_EXPOSE` and `HTTPS_EXPOSE` values, but then Mercure stops working in my Symfony app due to a 502 bad gateway / cors origin on the selected port 

Hence: I already tried changing the ports, and saw that it caused 502. So the info was already there, albeit part of the "wider" problem that traefik adds a non-configurable port that is already in use - which several replies here proposed to solve by "not using port 9999 in custom docker compose", on which I replied that this leads to traefik somehow causing a 502 on (in this case) mercure's url using that other-than-9999 port.

TLDR I am not sure which specific info you would then want me to repeat or elaborate on in a new issue, or even what title I should give the new issue. Thanks in advance for giving pointers.

**Comment by rfay** [comment]
What we need in a new issue is:
* With changed, non-conflicting ports
* What happens?
* How can it be reproduced?

It looks important, but it doesn't belong here, we've veered off, and this one has important things that have to be fixed. 

If you can reproduce in a trivial repo (just an index.php?) then please do that. If it takes mercure, please use a trivial setup and push it up to a github repo with a README showing how to demonstrate the problem. Please try to figure out if it's mercure-specific (likely).

**Comment by tomderoo** [comment]
Alright, I'll take a look at with a fresh, less sleepy head :) I still classify it as a bug, right? I'll make sure to link to this issue as an origin, too.

**Comment by rfay** [comment]
I consider it a bug until understood better. But until we can reproduce it will be very difficult to study. Again, thanks for raising the original issue here, it's important, and I imagine your other one is also, but I can't imagine that it's not application related. But we'll chase it with your help!

**Comment by tomderoo** [comment]
Alright, made new issue https://github.com/ddev/ddev/issues/5203 that included a sandbox project at https://github.com/tomderoo/sandbox-sf6-mercure-ddev - I know it has Symfony to frame Mercure, but I do not know how else to create that in a fast and reliabe manner...

**Comment by Awais-019** [comment]
I also ran into this issue on updating ddev, can anyone do anyone find the solution?

**Comment by rfay** [comment]
@Awais-019 you can't use port 9999 in your docker-compose.*.yaml. Use another port.

**Comment by amirinterlutions** [comment]
Fastest solution currently  :zany_face: 
Linux
`sudo apt install ddev=1.21.6`
Mac
`curl -fsSL https://ddev.com/install.sh | bash -s v1.21.6`

**Comment by rfay** [comment]
You can also just `ddev config global --router=nginx-proxy`, that's all you need to do to continue using the traditional router with DDEV v1.22.0. Must faster :)

**Comment by Awais-019** [comment]
That was really helpful thans

**Comment by rfay** [comment]
I tried to make a DDEV add-on that could fix this problem, but I don't think it can be done that way.

This should urgently get an override for the port used for Traefik management, for v1.22.1.

**Comment by rfay** [comment]
This should be very much sorted out in DDEV v1.22.1, just released. The port used is now 10999, and it's configurable.
